### PR TITLE
fix(gitlab): ensure `getPrList()` runtime integrity 

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
Update the runtime pr list when a pr is updated

## Context
Ref: #32040 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository (locally)

### CODE REVIEW

File: lib/modules/platform/gitlab/index.ts [hunk 1]
```diff
@@ -855,10 +855,39 @@ export async function updatePr({
...
-    { body },
-  );
+  const updatedPrInfo = (
+    await gitlabApi.putJson<{
+      iid: number;
+      source_branch: string;
+      title: string;
+      state: string;
+      created_at: string;
+    }>(`projects/${config.repository}/merge_requests/${iid}`, { body })
```
Comment (rarkins) [inline lib/modules/platform/gitlab/index.ts:865]: Can you reuse `GitLabMergeRequest` here instead?

File: lib/modules/platform/gitlab/index.ts [hunk 2]
```diff
@@ -855,10 +855,39 @@ export async function updatePr({
...
+  ).body;
+
+  const updatedPr: Pr = massagePr({
+    number: updatedPrInfo.iid,
+    state: updatedPrInfo.state === 'opened' ? 'open' : updatedPrInfo.state,
+    title: updatedPrInfo.title,
+    createdAt: updatedPrInfo.created_at,
+    sourceBranch: updatedPrInfo.source_branch,
+  });
```
Comment (rarkins) [inline lib/modules/platform/gitlab/index.ts:874]: Similar code to this seems duplicated throughout this gitlab module. Can any of it be consolidated?

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/modules/platform/gitlab/index.ts (hunk 1 above)
Comment (secustor) [inline lib/modules/platform/gitlab/index.ts:865]: Create a type for that generic. 

Can we use Zod here?

**Review by viceice — COMMENTED** [review]
see open conversations

File: lib/modules/platform/gitlab/index.ts (hunk 1 above)
Comment (RahulGautamSingh) [reply lib/modules/platform/gitlab/index.ts:865]: Yes, I think this is necessary can we can start with Gitlab as the types are messy here. A task for another PR though as we need to also handle the other functions like createPr, getPrList, getPr, getBranchPr, and findPr.

File: lib/modules/platform/gitlab/index.ts (hunk 2 above)
Comment (RahulGautamSingh) [reply lib/modules/platform/gitlab/index.ts:874]: https://github.com/renovatebot/renovate/pull/32428

**Comment by RahulGautamSingh** [comment]
Blocked by: https://github.com/renovatebot/renovate/pull/32428

File: lib/modules/platform/gitlab/index.ts [hunk 3]
```diff
@@ -811,10 +811,30 @@ export async function updatePr({
...
-    { body },
-  );
+  const updatedPrInfo = (
+    await gitlabApi.putJson<GitLabMergeRequest>(
+      `projects/${config.repository}/merge_requests/${iid}`,
+      { body },
+    )
+  ).body;
+
+  const updatedPr: Pr = prInfo(updatedPrInfo);
```
Comment (viceice) [inline lib/modules/platform/gitlab/index.ts:821]:
```suggestion
  const updatedPr = prInfo(updatedPrInfo);
```

**Review by viceice — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Comment by secustor** [comment]
@rarkins You have to approves as you have requested changes

**Review by rarkins — APPROVED** [review]

**Pull request merged**